tests/jeos/sccreg.pm

```diff
@@ -8,19 +8,29 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Register JeOS with SUSEConnect as prerequisite
-# Maintainer: Oliver Kurz <okurz@suse.de>
+# Summary: Register JeOS and install Server-like modules
+# Maintainer: Michal Nowak <mnowak@suse.com>
 
-use base "opensusebasetest";
+use base 'opensusebasetest';
 use strict;
 use testapi;
+use utils 'zypper_call';
 
 sub run {
-    my $sccmail = get_required_var('SCC_EMAIL');
     my $scccode = get_required_var('SCC_REGCODE');
-    my $url     = get_var('SCC_URL', 'https://scc.suse.com');
+    my $url     = get_required_var('SCC_URL');
+    my $arch    = get_required_var('ARCH');
+    my $version = get_required_var('VERSION');
 
-    assert_script_run "SUSEConnect --url=$url -e $sccmail -r $scccode";
+    # Register SLES
+    assert_script_run "SUSEConnect --url=$url -r $scccode";
```

Comment:
ok but we have other locations where we use SUSEConnect already. It would be appreciated if you can help to remove duplication in another PR

Reply:
I found `console/suseconnect_scc` test but that (1) installs modules in "random' order (hash is involved) but there are dependencies in between modules, it fails for me so I wonder if it works on installed SLES, and (2) it seems I am unable to install some modules on JeOS: http://assam.suse.cz/tests/662#step/sccreg/17. @yxususe Any idea?

Reply:
Hi Michal, I am not an expert with hash or JeOS, however in my local testruns, I didn't meet dependency problem between modules.

Reply:
There is dependency issue between modules: http://localhost/tests/1697#step/suseconnect_scc/12

Reply:
"Page not found" - but I believe I replied elsewhere to the same cause.

Reply:
@yxususe what @mnowaksuse refers to is that keys in a perl hash are not expected to be stored in alphabetical order. So `for my $module (keys %modules)` is not expected to return the list of modules in the order they have been entered. As there is a dependency between modules, e.g. "scripting" depends on "basesystem" this might induce problems